kotlin-inject-compiler/core/src/main/kotlin/me/tatarka/inject/compiler/TypeResultGenerator.kt

```diff
@@ -115,33 +115,53 @@ data class TypeResultGenerator(val options: Options, val implicitAccessor: Acces
             if (isProperty) {
                 add("%N", methodName)
             } else {
-                add("%N(", methodName)
-                parameters.forEachIndexed { i, param ->
+                add("%N", methodName)
+                val firstParam = parameters.firstOrNull()?.result
+                if (firstParam is TypeResult.Function) {
```

Comment:
Hm, not sure I love how much this has to know about it's parameters, I wonder if there's a way to improve this?

Reply:
Hmm looking at it again, it seems there's a bug. The intention was to use trailing lambda syntax for functions that only have a single lambda parameter, but if there are multiple params and the first one is a lambda, then it will only write out the first param :scream: 

That bug aside, I thought it was much more readable that way, but lambdas in general get weird because they have newlines but other params don't. What are your thoughts on that?

Reply:
Looking at it more, I'm going to remove that part. It's the difference between seeing:

```
bar({
  foo(fBar)
})
```

and 

```
bar {
  foo(fBar)
}
```

which I think for generated code is fine.

Reply:
Updated